include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,537 @@
+
+        if (it == anchors.begin())
+            return value_type((*sequence)[i]); // since no gaps happen before i
+
+        size_type gap_len{(*--it).second};
+
+        if (it != anchors.begin())
+            gap_len -= (*(std::prev(it))).second;
+
+        if (i < (*it).first + gap_len)
+           return gap{};
+        else
+           return value_type((*sequence)[i - (*it).second]);
+    }
+    //!\}
+
+    /*!\name Comparison operators
+     * \{
+     */
+    /*!\brief Compare seqan3::gap_decorator_anchor_set by underlying sequence and gaps.
+     * \tparam    inner_type2  The underlying type of the decorator to compare with; must be the same as inner_type
+     *                         when ignoring constness.
+     * \param[in] rhs          The right-hand side gap decorator to compare.
+     * \returns A boolean flag indicating (in)equality of the aligned sequences.
+     *
+     * \par Complexity
+     * Worst case: \f$O(n*\log k)\f$
+     *
+     * \par Exceptions
+     *
+     * No-throw guarantee. Does not modify the aligned sequences.
+     */
+    template <typename inner_type2>
+    //!\cond
+        requires std::is_same_v<std::remove_const_t<inner_type>, std::remove_const_t<inner_type2>>
+    //!\endcond
+    constexpr bool operator==(gap_decorator_anchor_set<inner_type2> const & rhs) const noexcept
+    {
+        if (anchors.size() != rhs.anchors.size())
+            return false;
+
+        if (sequence) // not nullptr
+        {
+            if (rhs.sequence) // not nullptr
+            {
+                if (sequence->size() == rhs.sequence->size())
+                    return std::ranges::equal(*this, rhs);
+            }
+        }
+        else if (!rhs.sequence)
+        {
+            return true; // both are nullptr (and both anchors should be empty as inserting gaps into a nullptr is UB)
+        }
+
+        return false;
+    }
+
+    //!\copydoc operator==
+    template <typename inner_type2>
+    //!\cond
+        requires std::is_same_v<std::remove_const_t<inner_type>, std::remove_const_t<inner_type2>>
+    //!\endcond
+    constexpr bool operator!=(gap_decorator_anchor_set<inner_type2> const & rhs) const noexcept
+    {
+        return !(operator==(rhs));
+    }
+    //!\}
+private:
+    //!\brief The gap type as a tuple storing position and accumulated gap lengths.
+    using gap_t = typename std::pair<size_t, size_t>;
+
+    /*!\brief Structure allowing the comparison of gaps.
+     * \tparam        gap_t Type of the gap storing position an length accumulator.
+     *
+     * \details
+     *
+     * It is assumed that the gap structure is always in a consistent
+     * state, i.e. there are no two gaps that are overlapping. The ordering of gaps
+     * structures is exclusively dependent on the gap starting position which is
+     * the ordering criterion for the anchor set implemented as ordered red-black tree.
+     */
+    template <typename gap_t>
```

Comment:
confusing usage of `gap_t` as a template parameter and as a member type before

Reply:
Also gap_t sounds like an alphabet, I would really use something different here. We need to have consistent naming throughout SeqAn. 

gap = single gap symbol
? = continuous stretch of gap symbols